kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js

```diff
@@ -120,16 +121,50 @@ define([
                 .on('reset', function() {
                     $dropzoneElem.find('#global-info').css({'display': 'none'});
                     $($dropzoneElem.find('#total-progress .progress-bar')).css({'width': '0%'});
+                    $('#clear-all-btn-container').remove();
+                    $('#clear-all-btn').remove();
                 })
                 .on('error', (err) => {
                     let errorText = 'unable to upload file!';
                     if (err && err.xhr && err.xhr.responseText) {
                         errorText = err.xhr.responseText;
                     }
                     $dropzoneElem.find('.error.text-danger').text('Error: ' + errorText);
+
+                    // Check to see if there already a button in the dropzone area
+                    if (!$dropzoneElem.find('#clear-all-btn').length){
+                        $dropzoneElem.append(this.makeClearAllButton());
+
+                    } else {
+                        // If there is a button already in the area, it has to be removed,
+                        // and appened to the new document when additional errored files are added.
+                        $('#clear-all-btn-container').remove();
+                        $('#clear-all-btn').remove();
```

Comment:
Couple questions/thoughts here: 

Why do we have to remake the button every time? Is there an issue with clearing them out properly if it doesn't get reinitialized? 

These remove statements may make more sense as part of makeClearAllButton function. That way if it ever got called in another context you won't have to duplicate the removal.

Reply:
Putting the remove functions in the reset block was to handle an edge case. If there is only a single file in the dropzone, and a user clicks the cancel button to remove the file instead of the clear all button, the clear all button would remain. I could make a delete button function and have the on click and the reset block call that instead of having to call each of them individually. 

We have to remake the button each time too append it to the end of the drop zone document. If there is an errored file in the dropzone, and another file is added after it, the button is in the middle of the list because the new file is just appended to the end of the document in the drop zone.

Reply:
Cool that all makes sense, thanks for explaining. Checked out your refactor and it's looking nice.